alfalfa_web/server/api.js

```diff
@@ -78,21 +62,24 @@ class AlfalfaAPI {
         const model = await this.models.findOne({ _id: run.model });
         let site = await this.sites.findOne({ ref_id: siteRef });
 
+        const site_dict = {
+          id: siteRef,
+          name: model.model_name,
+          status: run.status.toLowerCase(),
+          datetime: "",
+          simType: run.sim_type,
+          uploadTimestamp: run.created,
+          uploadPath: `uploads/${model.ref_id}/${model.model_name}`,
+          errorLog: run.error_log
+        };
+
         if (site) {
-          site = mapHaystack(site);
           const siteHash = await getHash(this.redis, siteRef);
 
-          return {
-            id: run.ref_id,
-            name: site?.dis,
-            status: run.status.toLowerCase(),
-            simType: site.sim_type,
-            datetime: siteHash?.sim_time ?? "",
-            uploadTimestamp: run.created,
-            uploadPath: `uploads/${model.ref_id}/${model.model_name}`,
-            errorLog: run.error_log
-          };
+          site_dict.name = site?.dis;
+          site_dict.datetime = siteHash?.sim_time;
```

Comment:
Since this could set the value to undefined let's use `siteHash?.sim_time ?? ""`.  Same for _name_, if `model.model_name` was set that could be replaced with undefined if `site.dis` isn't set